lib/ipmi_backend_utils.pm

```diff
@@ -70,79 +68,86 @@ sub setup_console_in_grub {
-                $cmd = "sed -i '/### END \\\/etc\\\/grub.d\\\/00_header ###/iset default=2' $grub_cfg_file";
-                assert_script_run($cmd);
-            }
-
-
-        }
-        elsif (${virt_type} eq "kvm") {
-            $bootmethod = "linux";
-            $search_pattern = "boot";
-        }
-        else {
-            die "Host Hypervisor is not xen or kvm";
         }
-
+    }
+    elsif (${virt_type} eq "kvm") {
         #enable Intel VT-d for SR-IOV test running on intel SUTs
         my $intel_option = "";
-        if (${virt_type} eq "kvm" && get_var("ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH") && script_run("grep Intel /proc/cpuinfo") == 0) {
+        if (get_var("ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH") && script_run("grep Intel /proc/cpuinfo") == 0) {
             $intel_option = "intel_iommu=on";
         }
 
         $cmd
-          = "cp $grub_cfg_file ${grub_cfg_file}.org "
-          . "\&\& sed -ri '/($bootmethod\\s*.*$search_pattern)/ "
+          = "sed -ri '/(linux\\s*.*boot)/ "
           . "{s/(console|loglevel|loglvl|guest_loglvl)=[^ ]*//g; "
-          . "/$bootmethod\\s*.*$search_pattern/ s/\$/ console=$ipmi_console,115200 console=tty loglevel=5 $intel_option/;}; "
-          . "s/timeout=-{0,1}[0-9]{1,}/timeout=30/g;"
+          . "/linux\\s*.*boot/ s/\$/ console=$ipmi_console,115200 console=tty loglevel=5 $intel_option/;}; "
           . "' $grub_cfg_file";
         assert_script_run($cmd);
-        save_screenshot;
-        $cmd = "sed -rn '/(multiboot|$bootmethod\\s*.*$search_pattern|timeout=)/p' $grub_cfg_file";
-        assert_script_run($cmd);
-        save_screenshot;
-
-        if (!script_run('grep HPE /sys/class/dmi/id/board_vendor') == 0) {
-            $cmd = "sed -ri '/^terminal.*\$/ {:mylabel; n; s/^terminal.*\$//;b mylabel;}' $grub_cfg_file";
-            assert_script_run($cmd);
-            $cmd = "sed -ri '/^[[:space:]]*\$/d' $grub_cfg_file";
-            assert_script_run($cmd);
-            $cmd = "sed -ri 's/^terminal.*\$/terminal_input console serial\\nterminal_output console serial\\nterminal console serial/g' $grub_cfg_file";
-            assert_script_run($cmd);
-        }
-        $cmd = "cat $grub_cfg_file $grub_default_file";
-        assert_script_run($cmd);
-        save_screenshot;
-        upload_logs($grub_default_file);
     }
     else {
-        die "Not supported grub version!";
+        die "Host Hypervisor is not xen or kvm";
+    }
+
+    if (!script_run('grep HPE /sys/class/dmi/id/board_vendor') == 0) {
+        $cmd = "sed -ri '/^terminal.*\$/ {:mylabel; n; s/^terminal.*\$//;b mylabel;}' $grub_cfg_file";
+        assert_script_run($cmd);
+        $cmd = "sed -ri '/^[[:space:]]*\$/d' $grub_cfg_file";
+        assert_script_run($cmd);
+        $cmd = "sed -ri 's/^terminal.*\$/terminal_input console serial\\nterminal_output console serial\\nterminal console serial/g' $grub_cfg_file";
+        assert_script_run($cmd);
```

Comment:
Could you help add comment for this block ?

Reply:
Actually I did not notice when and why these codes were added. This PR does not touch this part.